From 27a15664cd37c4d786c3b381636e7a674d372e0c Mon Sep 17 00:00:00 2001
From: Dan Carpenter <dan.carpenter@linaro.org>
Date: Fri, 14 Mar 2025 13:10:26 +0300
Subject: [PATCH 115/200] UPSTREAM: spi: sg2044-nor: fix uninitialized variable
 in probe

The "base" pointer is uninitialized.  It should be "spifmc->io_base"
instead.

Fixes: de16c322eefb ("spi: sophgo: add SG2044 SPI NOR controller driver")
Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
Reviewed-by: Tudor Ambarus <tudor.ambarus@linaro.org>
Link: https://patch.msgid.link/d343921b-16b8-429b-888a-f51bb6f2edc8@stanley.mountain
Signed-off-by: Mark Brown <broonie@kernel.org>
(cherry picked from commit a1d8f70954f69e333b252795808164839bb2e7cf)
Signed-off-by: Han Gao <rabenda.cn@gmail.com>
---
 drivers/spi/spi-sg2044-nor.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/spi-sg2044-nor.c b/drivers/spi/spi-sg2044-nor.c
index 5e68202b59fc..a59aa3fc55d2 100644
--- a/drivers/spi/spi-sg2044-nor.c
+++ b/drivers/spi/spi-sg2044-nor.c
@@ -428,7 +428,6 @@ static int sg2044_spifmc_probe(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 	struct spi_controller *ctrl;
 	struct sg2044_spifmc *spifmc;
-	void __iomem *base;
 	int ret;
 
 	ctrl = devm_spi_alloc_host(&pdev->dev, sizeof(*spifmc));
@@ -445,8 +444,8 @@ static int sg2044_spifmc_probe(struct platform_device *pdev)
 	spifmc->ctrl = ctrl;
 
 	spifmc->io_base = devm_platform_ioremap_resource(pdev, 0);
-	if (IS_ERR(base))
-		return PTR_ERR(base);
+	if (IS_ERR(spifmc->io_base))
+		return PTR_ERR(spifmc->io_base);
 
 	ctrl->num_chipselect = 1;
 	ctrl->dev.of_node = pdev->dev.of_node;
-- 
2.47.3

